test/integration_test/clusterdomain_test.go

```diff
@@ -0,0 +1,179 @@
+// +build integrationtest
+
+package integrationtest
+
+import (
+	"testing"
+
+	"github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/portworx/sched-ops/k8s"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/stretchr/testify/require"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	activateName   = "activate-cdu"
+	deactivateName = "deactivate-cdu"
+)
+
+var (
+	cdsName    string
+	domainMap  = map[string]bool{"dc1": true, "dc2": true}
+	domainList = []string{"dc1", "dc2"}
+)
+
+func testClusterDomains(t *testing.T) {
+	// Fetch the cluster domains
+	cdses, err := k8s.Instance().ListClusterDomainStatuses()
+	if err != nil || len(cdses.Items) == 0 {
+		t.Skip("Skipping cluster domain tests: %v", err)
+	}
+	cds := cdses.Items[0]
+	cdsName = cds.Name
+	var domains []string
+	for _, domain := range cds.Status.Active {
+		domains = append(domains, domain)
+	}
+
+	if len(domains) == 0 {
+		t.Skip("Skipping cluster domain tests: No cluster domains found")
+	}
+
+	t.Run("failoverAndFailbackTest", failoverAndFailbackClusterDomainTest)
+}
+
+func triggerClusterDomainUpdate(
+	t *testing.T,
+	name string,
+	domain string,
+	active bool,
+) {
+	setRemoteConfig("")
+
+	_, err := k8s.Instance().CreateClusterDomainUpdate(&v1alpha1.ClusterDomainUpdate{
+		ObjectMeta: meta.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.ClusterDomainUpdateSpec{
+			ClusterDomain: domain,
+			Active:        active,
+		},
+	})
```

Comment:
Need a VaildateClusterDomainUpdate here to make sure the update was successful

Reply:
The validate on the ClusterDomainStatus will also check that it's in the correct state but good to the check that status here too.

Reply:
done